pkg/utilities/components/kustomization/writer.go

```diff
@@ -55,16 +75,72 @@ func WriteFluxKustomization(kustomization *kustomizev1.Kustomization, fileName,
 }
 
 // WriteObjectsToFilesystem writes the given objects to the filesystem at the specified baseDir and filePathDir.
+// If the manifest file already exists, it patches changes from the new default.
+// Additionally, it maintains a default version of the manifest in a separate directory for future diff checks.
 func WriteObjectsToFilesystem(objects map[string][]byte, baseDir, filePathDir string, fs afero.Afero) error {
 	if err := fs.MkdirAll(path.Join(baseDir, filePathDir), 0700); err != nil {
 		return err
 	}
 
 	for fileName, object := range objects {
 		filePath := path.Join(filePathDir, fileName)
-		if err := meta.CreateOrUpdateManifest(object, baseDir, filePath, fs); err != nil {
+
+		filePathCurrent := path.Join(baseDir, filePath)
```

Comment:
As addressed offline: I have doubts coupling the Yaml Diff handling to the Kustomization utility, if we later need this functionality without Kustomize.

Does it make sense to extract this to a separate `file` package?